nautobot_netbox_importer/generator/fields.py

```diff
@@ -248,3 +256,53 @@ def define_disable(field: SourceField) -> None:
         field.disable(reason)
 
     return define_disable
+
+
+def auto_increment(prefix="", nautobot_name: FieldName = "") -> SourceFieldDefinition:
+    """Auto increment field value, if the source value is empty.
+
+    Use to set the field value to a unique auto incremented value.
+
+    Supports string and integer fields.
+
+    Args:
+        prefix (str): Optional prefix to be added to the auto incremented value. Valid for string fields only.
+        nautobot_name (str): Optional name for the Nautobot field.
+
+    Returns:
+        A function that defines an auto increment field importer.
+    """
+
+    def define_auto_increment(field: SourceField) -> None:
+        key = f"{field.wrapper.content_type}.{field.name}_prefix"
+
+        original_importer = field.set_importer(nautobot_name=nautobot_name)
+        if not original_importer:
+            return
+
+        if field.nautobot.internal_type in INTERNAL_INTEGER_FIELDS:
+            if prefix:
+                raise ValueError("Prefix is not supported for integer fields")
+
+        if field.nautobot.internal_type not in INTERNAL_STRING_FIELDS:
+            raise ValueError(f"Field {field.name} is not a string or integer field")
+
+        def auto_increment_importer(source: RecordData, target: DiffSyncBaseModel) -> None:
+            value = field.get_source_value(source)
+            if value not in EMPTY_VALUES:
+                original_importer(source, target)
+                return
+
+            if key not in _AUTO_INCREMENTS:
+                _AUTO_INCREMENTS[key] = 0
+            _AUTO_INCREMENTS[key] += 1
```

Comment:
Maybe we can declare ` _AUTO_INCREMENTS = collections.defaultdict(int)` online 30 and then here we only need to do.
```suggestion
            _AUTO_INCREMENTS[key] += 1
```